cmd/storage/postgresql/external.go

```diff
@@ -0,0 +1,197 @@
+package storage
+
+import (
+	"database/sql"
+	"errors"
+	"fmt"
+	"time"
+
+	add "github.com/Tanya1515/gophermarket/cmd/additional"
+)
+
+func (db *PostgreSQL) RegisterNewUser(user add.User) error {
```

Comment:
```suggestion
func (db *PostgreSQL) RegisterNewUser(ctx context.Context, user add.User) error {
```
Давай все функции переделаем на *Context и будем пробрасывать им контекст от хендлеров.